openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -31,6 +38,116 @@ def get_identity(self, data):
         except AttributeError:
             return None
 
+class AccountFullUserProfileReadOnlySerializer(serializers.Serializer):
+    """
+    Class that serializes the portion of User model needed for account information.
+    """
+    def __init__(self, *args, **kwargs):
+        # Don't pass the 'configuration' arg up to the superclass
+        self.configuration = kwargs.pop('configuration', None)
+        if not self.configuration:
+            self.configuration = settings.ACCOUNT_VISIBILITY_CONFIGURATION
+
+        # Don't pass the 'admin_fields' arg up to the superclass
+        self.admin_fields = kwargs.pop('admin_fields', False)
+
+        # Instantiate the superclass normally
+        super(AccountFullUserProfileReadOnlySerializer, self).__init__(*args, **kwargs)
+
+    def to_native(self, user):
+        """
+        Overwrite to_native to handle custom logic since we are serializing two models as one here
+        :param user: User object
+        :return: Dict serialized account
+        """
+        profile = user.profile
+
+        data = {
+            "username": user.username,
+            "url": ''.join([
```

Comment:
Note: it took me a little double checking to verify that this "url" parameter doesn't get included in the User API endpoint. This would be a problem as it is a link back to itself. The trick is that "url" isn't listed in the configurable list of admin fields in ``common.py``. It might be worth a comment to this effect.

I was expecting that the URL would be added only by the place that wants it (i.e. in the expand serializer). Would it be possible to have a dedicated subclass just for this? It seems strange to have a field that most callers don't want and that is filtered out by default.